Editor/EditorCore/MenuAction.cs

```diff
@@ -246,11 +248,31 @@ protected virtual MenuActionState optionsMenuState
             get { return MenuActionState.Hidden; }
         }
 
+        public static UnityAction<MenuAction> actionChanged;
+        public virtual void OnActionChanged(MenuAction action)
+        {
+            // Debug.Log("Action Changed");
+            // if(this.Equals(action))
+            // {
+            //     Debug.Log("Action changed to me : "+action.GetType());
+            // }
+        }
+
+        /// <summary>
+        /// Perform whatever action this menu item is supposed to do. You are responsible for implementing Undo.
+        /// </summary>
+        /// <returns>A new ActionResult with a summary of the state of the action's success.</returns>
+        public ActionResult DoAction()
+        {
+            actionChanged(this);
+            return DoAction_Internal();
+        }
+
```

Comment:
I have a vague memory of discussing this, but can you remind me why it is necessary to add this event? Additionally, this will need to go through the a deprecation phase before changing the signature.

Reply:
The event was added because the toggle tools (polyshape, cut tool, shape tool), need to be aware of probuilder menuaction changes so that it can disable a toggle tool when selecting a new PB action. 
If you have another idea to make this, it could be better :)

I was reusing the method DoAction, thinking that might be transparent, but indeed a deprecation phase would be better if this structure is kept.